### PR TITLE
Yaa development2

### PR DESCRIPTION
**Pull request by yaaowusu00**
Finished implementing translation feature and also cleaned up the UI such that the comments are displayed in a box using bullets, rather than as a JSON. I decided to keep the English comments on the page while displaying the translated version, instead of just displaying the translated comments.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -34,48 +37,67 @@
...
   }
    
   @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {  
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {   
     // Get the input from the form, add comment to the ArrayList
      String  userComment = getuserComment(request);
-     msgs.add(userComment);
+     
+    if ((userComment != null) || (userComment!= "")) {
```
Comment (ctrokid) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: The `(userComment != ..)` parentheses are unnecessary I believe.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -34,48 +37,67 @@
...
+    else {
+      Translate translate = TranslateOptions.getDefaultInstance().getService();
+      //iterate through all the comments and translate them 
+      for (Entity entity : results.asIterable()) {
+         String com  = (String)entity.getProperty("string");
+         Translation translation =
+         translate.translate(com, Translate.TranslateOption.targetLanguage(code));
+         String translatedText = translation.getTranslatedText();
+         comList.add(translatedText);
+        }
```
Comment (ctrokid) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: nit: formatting

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -34,48 +37,67 @@
...
-    Gson gson = new Gson();
+    else {
+      Translate translate = TranslateOptions.getDefaultInstance().getService();
+      //iterate through all the comments and translate them 
+      for (Entity entity : results.asIterable()) {
+         String com  = (String)entity.getProperty("string");
+         Translation translation =
+         translate.translate(com, Translate.TranslateOption.targetLanguage(code));
+         String translatedText = translation.getTranslatedText();
+         comList.add(translatedText);
```
Comment (ctrokid) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: nit: you could just do `comList.add(translation.getTranslatedText());`

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -29,8 +29,34 @@ function addRandomFact() {
...
 function getMessage() {
-  fetch('/data').then(response => response.text()).then((messages) => {
-    document.getElementById('comment').innerHTML = messages;
-  });
+  const commentContainer = document.getElementById('comment');
+  fetch('/data').then(response => response.json()).then((messages) => {
+     messages.forEach((line) => {
+         commentContainer.appendChild(newLiElement(line));
+     });
+    });
```
Comment (ctrokid) [inline portfolio/src/main/webapp/script.js]: nit: formatting

**Comment by yaaowusu00** [comment]
I was about to open a PR for this week's stuff and just realized I never fixed last weeks. Sorry about that!

**Pull request merged**